src/Endpoints/Delegates/HandlesSystem.php

```diff
@@ -4,8 +4,18 @@
 
 namespace Meilisearch\Endpoints\Delegates;
 
+use Meilisearch\Endpoints\Health;
+use Meilisearch\Endpoints\Stats;
+use Meilisearch\Endpoints\TenantToken;
+use Meilisearch\Endpoints\Version;
+
 trait HandlesSystem
 {
+    protected Health $health;
```

Comment:
To understand it entirely, since the traits are a mechanism for code reuse, we are now defining the properties that are used inside of the trait internally instead of simply waiting for the class that will `use Trait` to define them, right?

If the statement above is right, why defining this in such a way is even possible? Do you know any history behind this language decision?

Reply:
This change is to make DX better, while trait is for code sharing, they would fail badly when you don't declare required properties for them in the class that uses them (previous version of code), while with `protected` (which means shared within inheritance but still not publicly visible) means that developer can use methods by just adding those traits with not further changes.

In fact, those traits in this SDK are overkill IMHO, cause the code here is not really so abstract to be shared as much as it seems, cause those traits in fact are definitions of endpoints, which will not really change over time, yet they will not be shared outside of their scopes (you will not call 2-3 endpoints from 2-3 traits in same SDK method).

Reply:
Perfect, thanks for the explanation!